src/tests/ftest/container/fill_destroy_loop.py

```diff
@@ -0,0 +1,104 @@
+#!/usr/bin/python3
+"""
+  (C) Copyright 2019-2022 Intel Corporation.
+
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+"""
+import random
+
+from apricot import TestWithServers
+from general_utils import get_random_bytes, DaosTestError
+from test_utils_container import TestContainerData
+
+
+class BoundaryPoolContainerSpace(TestWithServers):
+    """Test class for pool Boundary testing.
+
+    Test Class Description:
+        Test to create a pool and container and write random data to fill the container and delete
+        container, repeat the test 100 times as boundary condition.
+
+    :avocado: recursive
+    """
+
+    DER_NOSPACE = "-1007"
+
+    def write_pool_until_nospace(self):
+        """write pool and container until pool is full. """
+        cont_obj_class = self.params.get("obj_class", "/run/container/*")
+        block_size = self.params.get("block_size", "/run/container/*")
+        akey_size = self.params.get("akey_size", "/run/container/*")
+        dkey_size = self.params.get("dkey_size", "/run/container/*")
+        # select a random data size based on block_size
+        data_size = random.randint(block_size*0.5, block_size*1.5) # nosec
+
+        container = self.get_container(self.pool)
+        container.open()
+        written_byte = 0
+        write_count = 1
+        data_str = get_random_bytes(data_size)
+        space_avail_to_write = True
+        while space_avail_to_write:
+            dkey = get_random_bytes(dkey_size)
+            akey = get_random_bytes(akey_size)
+            try:
+                written_byte += data_size
+                self.d_log.debug("--{0} writing cont-obj {1}  byte data, total {2}..".format(
+                    write_count, data_size, written_byte))
+                container.written_data.append(TestContainerData(False))
+                container.written_data[-1].write_record(
+                    container, akey, dkey, data_str, obj_class=cont_obj_class)
+                self.d_log.debug("--{0}wrote cont-obj, sz {1}".format(write_count, data_size))
+                write_count += 1
+            except DaosTestError as excep:
+                space_avail_to_write = False
+                if self.DER_NOSPACE in repr(excep):
+                    self.log.info("--(3)written_byte= %s, Der_no_space %s detected, pool is "
+                                  "unable for an additional %s byte object", written_byte,
+                                  self.DER_NOSPACE, data_size)
+                    free_space = self.pool.get_pool_free_space()
+                    self.log.info("--(4)free_space when pool is fulfill= %s", free_space)
+                else:
+                    self.fail("#Exception while writing object: {}".format(repr(excep)))
```

Comment:
Let's use the methods available in the TestContainer class and akey_size, dkey_size, and data_size test yaml parameters:
```suggestion
        container = self.get_container(self.pool)

        # Select a random data size based on block_size
        new_data_size = random.randint(container.data_size.value * 0.5, container.data_size.value * 1.5)   # nosec
        container.data_size.update(new_data_size, "data_size")

        while(1):
            try:
                container.write_objects()
            except DaosTestError as error:
                if self.DER_NOSPACE in error:
                    self.log.info(
                        "--(3) Der_no_space %s detected, pool is unable for an additional %s byte object",
                        self.DER_NOSPACE, container.data_size.value)
                    free_space = self.pool.get_pool_free_space()
                    self.log.info("--(4)free_space when pool is fulfill= %s", free_space)
                    break
                else:
                    self.fail("#Exception while writing object: {}".format(repr(excep)))
```